### PR TITLE
Feature/public interface docs

### PR DESCRIPTION
**Pull request by mdpye**
Introduces Jazzy config to group the docs by functional area and add non-type-derived overview docs.

*Important:* after merge, Jazzy should be invoked `jazzy --config .jazzy.conf` to update the docs in future.

### CODE REVIEW

**Comment by mdpye** [comment]
@zmarkan this PR adds some overview of initialisation and concepts of view models and providers - would it have helped you get started?

File: docs-src/View Models and Providers.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+contents of a UI component.
+
+For example, the `TypingUsersViewModel`
+
+- exposes a `String?` describing which users are typing in a given `Room`,
+  - each user is represented by the `name` field from the `User` object,
+    - if the `name` field is blank, the value `Anonymous user` is used in
+      place,
+  - the list of users is sorted by their name, and the list is truncated at a
+    certain number of users to say `... and other users`,
```
[the remark this answers is not in the document]
Comment (mdpye) [reply docs-src/View Models and Providers.md]: Heh, I did wonder if I was getting ahead of myself at the time. Let's add it, Zan questioned it during the testing as well.

[the remark this answers is not in the document]
Comment (grzesiekko) [reply docs-src/View Models and Providers.md]: OK

File: docs-src/Initialization.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+# Next
+
+Now that the SDK is ready for use, you will probably want to access chat data
+using either ViewModels or Providers which the SDK offers.
+
+You can also explore the methods offered by the `Chatkit` entry point object.
+Now that this object is initialized and connected, it is your handle to all
+further interation with the SDK.
```
[the remark this answers is not in the document]
Comment (mdpye) [reply docs-src/Initialization.md]: "Interaction" can be both a countable and uncountable noun, with differing plural forms (uncountable: "interaction", countable: "interactions"). This was intended to be the uncountable form. Yay English!

[the remark this answers is not in the document]
Comment (grzesiekko) [reply docs-src/Initialization.md]: Nevertheless the letter "c" is missing ;)

[the remark this answers is not in the document]
Comment (mdpye) [reply docs-src/Initialization.md]: :man_facepalming:

**Pull request merged**